sourcegen-plugin-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/gradle/builder/GradleTaskBuilder.java

```diff
@@ -0,0 +1,316 @@
+        return Type.GRADLE_TASK;
+    }
+
+    @Override
+    @NonNull
+    public List<ObjectDef> build(GradlePluginConfig pluginConfig) {
+        List<ObjectDef> objects = new ArrayList<>();
+        for (GradleTaskConfig taskConfig: pluginConfig.tasks()) {
+            objects.addAll(buildTask(pluginConfig.packageName(), taskConfig));
+        }
+        return objects;
+    }
+
+    private List<ObjectDef> buildTask(String packageName, GradleTaskConfig taskConfig) {
+        String taskType = packageName + "." + taskConfig.namePrefix() + TASK_SUFFIX;
+        ClassDefBuilder builder = ClassDef.builder(taskType)
+            .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+            .superclass(ClassTypeDef.of("org.gradle.api.DefaultTask"))
+            .addAnnotation("org.gradle.api.tasks.CacheableTask")
+            .addJavadoc(taskConfig.taskJavadoc());
+        builder.addInnerType(createWorkAction(taskConfig));
+        builder.addInnerType(createWorkActionParameters(taskConfig));
+        builder.addInnerType(createWorkActionParameterConfigurator(TypeDef.of(taskType), taskConfig));
+        builder.addInnerType(createClasspathConfigurator(TypeDef.of(taskType), taskConfig));
+
+        for (ParameterConfig parameter: taskConfig.parameters()) {
+            MethodDefBuilder propBuilder = MethodDef
+                .builder("get" + NameUtils.capitalize(parameter.source().getName()))
+                .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+                .addJavadoc(parameter.javadoc())
+                .returns(createGradleProperty(parameter));
+            if (parameter.output()) {
+                if (parameter.source().getType().isAssignable(File.class)) {
+                    if (parameter.directory()) {
+                        propBuilder.addAnnotation(AnnotationDef.builder(ClassTypeDef.of("org.gradle.api.tasks.OutputDirectory")).build());
+                    } else {
+                        propBuilder.addAnnotation(AnnotationDef.builder(ClassTypeDef.of("org.gradle.api.tasks.OutputFile")).build());
+                    }
+                }
+            } else {
+                propBuilder.addAnnotation("org.gradle.api.tasks.Input");
+                if (parameter.source().getType().isAssignable(File.class)) {
+                    if (parameter.directory()) {
+                        propBuilder.addAnnotation(AnnotationDef.builder(ClassTypeDef.of("org.gradle.api.tasks.InputDirectory")).build());
+                    } else {
+                        propBuilder.addAnnotation(AnnotationDef.builder(ClassTypeDef.of("org.gradle.api.tasks.InputFile")).build());
+                    }
+                    propBuilder.addAnnotation(AnnotationDef.builder(ClassTypeDef.of("org.gradle.api.tasks.PathSensitive"))
+                        .addMember("value", ClassTypeDef.of("org.gradle.api.tasks.PathSensitivity")
+                            .getStaticField("NONE", TypeDef.of("org.gradle.api.tasks.PathSensitivity"))
```

Comment:
:x: This should probably be configurable. No path sensitivity means that it doesn't matter if the file name changes, or the path to the files change, which is unlikely.

Reply:
You are right, I copied from some unusual property I saw